lib/ansible/plugins/become/sudo.py

```diff
@@ -136,4 +149,8 @@ def build_become_command(self, cmd, shell):
             except AttributeError as ex:
                 raise AnsibleError(f'The {shell._load_name!r} shell plugin does not support sudo chdir. It is missing the {ex.name!r} attribute.')
 
+        selinux_role = self.get_option('selinux_role')
+        if selinux_role:
+            flags += '-r %s' % (selinux_role)
```

Comment:
f-string, and these should all probably be `shlex.quote` or at least statically-embedded quotes, since we they'll be set dynamically

```suggestion
            flags += f'-r {shlex.quote(selinux_role)}'
```